File: sqlite3/fsharp/read/sqlite3_read.fs
// -------------------------------------------------------------------
//
//	sqlite3_read.fs
//
//					Feb/08/2016
//
// -------------------------------------------------------------------
open System
open System.Data
open Mono.Data.Sqlite

// -------------------------------------------------------------------
[<EntryPoint>]
let main (args : string[] ) =
  printfn "*** 開始 ***"

  let file_sqlite3 = args.[0]
  Console.WriteLine ("file_sqlite3 = " + file_sqlite3)

  let string_connect = "Data Source = " + file_sqlite3 + ";UTF8Encoding=True;Version=3"
  let conn = new SqliteConnection (string_connect)
  conn.Open()

  let cmd = new SqliteCommand( "SELECT * FROM cities", conn )
  let reader = cmd.ExecuteReader()

  while reader.Read()
    do System.Console.Write (reader.GetString 0 )
       System.Console.Write ("\t")
       System.Console.Write (reader.GetString 1 )
       System.Console.Write ("\t")
       System.Console.Write (reader.GetInt32 2 )
       System.Console.Write ("\t")
       System.Console.WriteLine (reader.GetString 3 )
       done

  reader.Close()

  printfn "*** 終了 ***"

  0

// -------------------------------------------------------------------

File: sqlite3/csharp/read_fill/sqlite3_read.cs
/* ------------------------------------------------------------------- */
/*
	/sqlite3_read.cs


					Jul/15/2010

*/
/* ------------------------------------------------------------------- */
using	System;
using	System.Data;
using	Mono.Data.Sqlite;

/* ------------------------------------------------------------------- */
class sqlite3_read
{
/* ------------------------------------------------------------------- */
static void Main (string[] args)
{
	Console.WriteLine ("*** 開始 ***");

	string file_sqlite3 = args[0];

	string str_connect = @"Data Source = " + file_sqlite3
			+ ";UTF8Encoding=True;Version=3";
 
	string command = "select ID,Name,Population,date_mod from cities";

	SqliteDataAdapter da_adapter
			= new SqliteDataAdapter (command,str_connect);

	DataTable dtable = new DataTable ();

	da_adapter.Fill (dtable);

	table_manipulate.display_proc (dtable);

	Console.WriteLine ("*** 終了 ***");
}
 
/* ------------------------------------------------------------------- */
}
/* ------------------------------------------------------------------- */

